### ISSUE TITLE
Failure when converting Gif w/RGBA palette to 'P' mode

### ISSUE THREAD

**Issue by wiredfool** [python-pillow/Pillow#2433]
Hoisting this from the bottom of #646, as it's actually another issue. Fundamentally, there's a weird palette thing going on. 

There's an image https://cdn.technologynetworks.com/tn/images/adverts/malvern-inline-rectangle-banner-31st-jan-2017.gif which causes failures in master / 4.0.0 and back a ways when doing:
```
im = Image.open(path)
im.save('temp.gif', save_all=True)
```

The save loop is fixed in #2374:  'fix' is this line: https://github.com/python-pillow/Pillow/pull/2374/commits/951e628aaa0b3b503f561b0f1a99364c12ec2f72#diff-0dca57e5083a5edca008f241582088f5R297  (verified by checking effectively the same change on master) Where instead of converting it to `P`, we're just copying the image/frame. 

So, this was accidentally fixed.

It looks like the underlying issue is as simple to trigger as:
```
from PIL import Image
im = Image.open('issue_646.gif')
im_p = im.convert('P')
im_p.load()  # boom
```

However, this does not trigger with `hopper.gif`, so it looks like we need an `RGBA` palette to trigger this.  If the palette mode is `RGBA`, and the palette is dirty, we get the failure.  

It doesn't look like the Gif _open/load is the issue, as it doesn't do anything that contains an `RGBA` palette, it's all `RGB`.

```
>>> im.palette.__dict__
{'rawmode': 'RGB', 'palette': '\xdbB\x03...\xff\xff\xff', 'dirty': 1, 'colors': {}, 'mode': 'RGB'}
>>> im.load()
<PixelAccess object at 0x7f3835c48970>
>>> im.palette.__dict__
{'rawmode': None, 'palette': '\xdbB\x03...\xff\xff\xff', 'dirty': 0, 'colors': {}, 'mode': 'RGBA'}
>>> im.info
{'background': 255, 'extension': ('XMP DataXMP', 814), 'version': 'GIF89a', 'transparency': 255, 'duration': 1500, 'loop': 0}
```

So, load is giving us an `RGBA` palette from an `RGB` palette and a transparency index.

```
>>> im2 = Image.open('Tests/images/hopper.gif')
>>> im2.info['transparency'] = 255
>>> im2.palette.__dict__
{'rawmode': 'RGB', 'palette': '\x0c\x0c\r\t...\xaf\xc5\xd1', 'dirty': 1, 'colors': {}, 'mode': 'RGB'}
>>> im2.load()
<PixelAccess object at 0x7f3835c48910>
>>> im2.palette.__dict__
{'rawmode': None, 'palette': '\x0c\x0c\r\t...\xaf\xc5\xd1', 'dirty': 0, 'colors': {}, 'mode': 'RGBA'}
>>> im2.show()
>>> im2_p = im2.convert('P')
>>> im2_p.palette.__dict__
{'rawmode': None, 'palette': bytearray(b';\x0c\r\t...\xaf\xc5\xd1'), 'colors': {(59, 113, 209): 0}, 'mode': 'RGBA', 'dirty': 1}
>>> im2_p.load()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "build/bdist.linux-x86_64/egg/PIL/Image.py", line 763, in load
ValueError: unrecognized raw mode
```

Repro'd with hopper.gif. 

The bug is either in:
 - `load()`, where it's converting the palette to `RGBA`, but it's still 768 bytes.
 - `convert('P')`, where it's taking an image already in `P` and modifying the palette.

**Comment by radarhere** [comment]
Trying the last code segment, this looks like it has been fixed now. It stops being reproducible after #2767

**Comment by hugovk** [comment]
Good to hear! Shall we add a test to guard against regressions?

**Comment by radarhere** [comment]
Okay, I've created a PR. I'll note that `im.load()` changes the palette mode from RGB to RGBA - the test fails with or without this line, but in order to preserve the idea of this PR, it is kept from the code provided here.
